### PR TITLE
Cleaning and extract image comparison

### PR DESCRIPTION
**Pull request by BielinskiLukasz**
Hello :)
I work with your code this week, great job. I made some cleaning there (removing doubles etc.). I extracted image comparison methods from ImageProcessor to a new class.

commit 6de649e
I'm not sure that the line is correct:
_viewportHeight -= scrollBarMaxWidth;

### CODE REVIEW

File: src/main/java/com/assertthat/selenium_shutterbug/core/PageSnapshot.java [hunk 1]
```diff
@@ -51,7 +51,7 @@ public PageSnapshot highlight(WebElement element) {
      */
     public PageSnapshot highlight(WebElement element, Color color, int lineWidth) {
         try {
-            image = ImageProcessor.highlight(image, new Coordinates(element, devicePixelRatio), color, lineWidth);
+            ImageProcessor.highlight(image, new Coordinates(element, devicePixelRatio), color, lineWidth);
```
Comment (glib-briia) [inline src/main/java/com/assertthat/selenium_shutterbug/core/PageSnapshot.java]: Need to keep assignment to image so subsequent chained methods can use updated image.

File: src/main/java/com/assertthat/selenium_shutterbug/core/PageSnapshot.java [hunk 2]
```diff
@@ -93,7 +93,7 @@ public PageSnapshot highlightWithText(WebElement element, Color elementColor, in
        try {
            highlight(element, elementColor, 0);
            Coordinates coords = new Coordinates(element, devicePixelRatio);
-           image = ImageProcessor.addText(image, coords.getX(), coords.getY() - textFont.getSize() / 2, text, textColor, textFont);
```
Comment (glib-briia) [inline src/main/java/com/assertthat/selenium_shutterbug/core/PageSnapshot.java]: Need to keep assignment to image so subsequent chained methods can use updated image.

File: src/main/java/com/assertthat/selenium_shutterbug/core/Snapshot.java [hunk 3]
```diff
@@ -176,7 +185,7 @@ public T withThumbnail(double scale) {
      * @return instance of type Snapshot
      */
     public T monochrome() {
-        this.image = ImageProcessor.convertToGrayAndWhite(this.image);
```
Comment (glib-briia) [inline src/main/java/com/assertthat/selenium_shutterbug/core/Snapshot.java]: Need to keep assignment to image so subsequent chained methods can use updated image.

**Comment by BielinskiLukasz** [comment]
Thanks for quick response and code review.

What about @6de649e commit? I didn't test other combination but there is assignment scrollBarMaxWidth to _viewportHeight and next again scrollBarMaxWidth to _viewportWidth. I am wondering about the multiple uses of width. If it's correct, there is TODO what I created, should be removed :)

**Comment by glib-briia** [comment]
Hi @BielinskiLukasz ,

Regarding @6de649e, I think we should leave this as is. scrollBarMaxWidth  is standard so it is used for both - removing vertical and horizontal scrollbars if present.

**Pull request merged**